synapse/config/registration.py

```diff
@@ -31,6 +31,7 @@ def read_config(self, config):
                 strtobool(str(config["disable_registration"]))
             )
 
+        self.registrations_require_3pid = config.get("registrations_require_3pid", [])
```

Comment:
This doesn't seem to affect just registrations and could be misleading to some people.

Reply:
this convo ended up in IRL as dave presumably missed this comment: the impact is:
 * There are three request tokens atm for verifying 3PIDs:
   * registration
   * password reset
   * adding a 3pid to your account for use in login (which in turn tries to go bind it on the IS if asked)
  * this PR restricts the format of all three with allowed_local_3pids
  * it does not limit IS bindings (other than in a weak manner given in practice the account/3pid api is used by clients today to trigger bindings)